ChromaticHQ/docker-images
# Create new php-apache-nvm service

## Description

This PR adds a new service called `php-apache-nvm`, to be consumed by projects that aim to manage their Node.js version with nvm.

## Motivation / Context

We would like to declare our Node.js version for a given project using a single canonical declaration: an `.nvmrc` file at the root of the project. This PR creates a whole new Docker image, as opposed to modifying our existing `php-apache` image, because it lowers the stakes. Once we have moved all projects that consume `php-apache` to use `php-apache-nvm` instead, we can simply retire the `php-apache` image altogether.

## Testing Instructions / How This Has Been Tested

An earlier version of this PR modified our `build-images` action to automatically build `php-apache-nvm-test` images and [push them to DockerHub](https://hub.docker.com/r/chromatichq/php-apache-nvm-test/tags). Once I ironed out all the details there, I restored `build-images.yml` and updated the `image-build-php` job to include `php-apache-nvm` as an additional service in both the validation step and the build step. The validation checks in this PR [show that this works as expected](https://github.com/ChromaticHQ/docker-images/actions/runs/1363402755).

In addition to the above, I have [a Draft PR in the CHQ repo](https://github.com/ChromaticHQ/chromatichq.com/pull/3169) which, for the time being, uses the aforementioned `php-apache-nvm-test` image that is still available in DockerHub. As the [Tugboat preview for that PR shows](https://dashboard.tugboat.qa/6152015e7207f2268c2becc4), we are able to use the nvm-installed version of Node.js (and Yarn), albeit with a few modifications to our Tugboat config.

I believe all of the above sufficiently validate that the changes introduced in this PR accomplish our goal without introducing any regressions.

## Screenshots

n/a

## Documentation

n/a — but I’ve left a few PR comments explaining a few things.
